pkg/sysfs/system.go

```diff
@@ -635,13 +669,38 @@ func (sys *system) discoverCPUs() error {
 		sys.Error("failed to get set of isolated cpus: %v", err)
 	}
 
+	sys.coreKindCPUs = make(map[CoreKind]idset.IDSet)
+	for kind, entry := range coreKindCPUPath {
+		cpus := idset.NewIDSet()
+		_, err = readSysfsEntry(sys.path, entry, &cpus)
+		if err != nil {
+			sys.Error("failed to get set of %s cpus: %v", kind, err)
+			if kind == PerformanceCore {
+				cpus = sys.onlineCPUs.Clone()
+			}
```

Comment:
minor nit: in case of any error, should we bail out clearly with smth like

```golang
    sys.coreKindCPUs = map[CoreKind]idset.IDSet{PerformanceCore: sys.onlineCPUs.Clone()}
```

Otherwise we could theoretically end up with overlapping set. Also would make the code more explicit. WDYT?

Reply:
Also, should we consider the possibility of a 3rd core kind being added, like "atom-low-power" á la Meteor Lake. Basically ensure that every CPU gets assigned to some kind? Can be left as a future exercise (with a note?)

Reply:
> minor nit: in case of any error, should we bail out clearly with smth like
> 
> ```go
>     sys.coreKindCPUs = map[CoreKind]idset.IDSet{PerformanceCore: sys.onlineCPUs.Clone()}
> ```
> 
> Otherwise we could theoretically end up with overlapping set. Also would make the code more explicit. WDYT?

https://github.com/containers/nri-plugins/pull/282 adds sanity checks for the detected CoreKinds, including a check for non-overlapping sets.

Reply:
> Also, should we consider the possibility of a 3rd core kind being added, like "atom-low-power" á la Meteor Lake. Basically ensure that every CPU gets assigned to some kind? Can be left as a future exercise (with a note?)

Definitely we should add it. I didn't know it's a thing already...

Reply:
I don't think that exists (haven't checked, though). That was just a note that maybe in the (not so distant) future we see more than two types

Reply:
@marquiz I reworked the patch series excluding anything related to hybrid architectures from this PR. It now only has the cache detection, cluster detection and clustered allocation bits.

I moved all infra hybrid architecture discovery bits and the related policy-specific bits to #282. Now I'm also considering splitting that into two parts: the infra bits and policy-specific bits. WDYT ?